File: kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/Environment.java
package kin.sdk;

import static kin.sdk.Utils.checkNotEmpty;

import kin.base.Network;

/**
 * Provides blockchain network details
 */
public class Environment {

    public static final Environment PRODUCTION =
		new Environment("https://horizon.kinfederation.com",
			"Kin Mainnet ; December 2018");

    public static final Environment TEST =
        new Environment("https://horizon-testnet.kininfrastructure.com/",
            "Kin Testnet ; December 2018");

    private final String networkUrl;
    private final Network network;

    /**
     * Build an Environment object.
     * @param networkUrl the URL of the blockchain node.
     * @param networkPassphrase the network id to be used.
     */
    public Environment(String networkUrl, String networkPassphrase) {
        checkNotEmpty(networkUrl, "networkUrl");
        checkNotEmpty(networkPassphrase, "networkPassphrase");
        this.networkUrl = networkUrl;
        this.network = new Network(networkPassphrase);
    }

    /**
     * Returns the URL of the blockchain node.
     */
    @SuppressWarnings("WeakerAccess")
    final public String getNetworkUrl() {
        return networkUrl;
    }

    /**
     * Returns the network id.
     */
    @SuppressWarnings("WeakerAccess")
    final public String getNetworkPassphrase() {
        return network.getNetworkPassphrase();
    }


    final public boolean isMainNet() {
        return PRODUCTION.getNetworkPassphrase().equals(network.getNetworkPassphrase());
    }

    final Network getNetwork() {
        return network;
    }

}

File: kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/Environment.kt
package kin.sdk

import kin.base.Network

/**
 * Provides blockchain network details
 */
class Environment
/**
 * Build an Environment object.
 *
 * @param networkUrl        the URL of the blockchain node.
 * @param networkPassphrase the network id to be used.
 */
(
    /**
     * Returns the URL of the blockchain node.
     */
    val networkUrl: String,
    networkPassphrase: String
) {
    companion object {

        val PRODUCTION = Environment("https://horizon.kinfederation.com",
            "Kin Mainnet ; December 2018")

        val TEST = Environment("https://horizon-testnet.kininfrastructure.com/",
            "Kin Testnet ; December 2018")

        private fun checkNotEmpty(string: String?, paramName: String) {
            require(!(string == null || string.isEmpty())) { "$paramName cannot be null or empty." }
        }
    }

    val network: Network

    /**
     * Returns the network id.
     */
    val networkPassphrase: String
        get() = network.networkPassphrase

    val isMainNet: Boolean
        get() = PRODUCTION.networkPassphrase == network.networkPassphrase

    init {
        checkNotEmpty(networkUrl, "networkUrl")
        checkNotEmpty(networkPassphrase, "networkPassphrase")
        this.network = Network(networkPassphrase)
    }
}
